### ISSUE TITLE
OOC DMatrix and set functions

### ISSUE THREAD

**Issue by jamescolless** [dmlc/xgboost#6541]
If I instantiate an OOC dmatrix with the following:

dtrain = xgb.DMatrix("train.csv?format=csv&label_column=0#dtrain.cache")

Where my label column is column 0 of the CSV and I write another file of weights to disk called "train.csv.weight", as discussed in (https://xgboost.readthedocs.io/en/latest/tutorials/input_format.html).

What is the effect of then calling:

dtrain.set_label(some_label_array)
dtrain.set_weight(some_weight_array)
dtrain.set_info(feature_weights=some_array_of_feature_weights)

None of these commands actually fail, but I suspect that the first two don't actually do anything, as the label column is defined as the 0th column of the on disk csv already, and the weights come from the .csv.weight file? If that's true we should probably raise an error when the user attempts to set these properties?

I'm not sure if the last command should work, although I guess it probably should?

Also, the documentation seems to suggest that feature_weights is only used in the sklearn api .fit call (https://xgboost.readthedocs.io/en/latest/python/python_api.html#module-xgboost.core) - is this correct?

**Comment by trivialfis** [comment]
You can get those meta info back to Python, see examples in `tests/python/test_dmatrix.py`, some tests like `test_feature_weights` `test_get_info` might help.
